src/kafka/protocol/api_versions_response.hh

```diff
@@ -28,12 +28,21 @@ namespace seastar {
 
 namespace kafka {
 
+struct unsupported_version_exception : public std::exception {
+    [[nodiscard]] const char *what() const noexcept override {
+        return "The client is not compatible with broker.";
+    }
+};
+
```

Comment:
I think, based on the usage below, that it would be handy to make this derive from std::runtime_exception and therefore allow for customizing the message based on the context (e.g. the comments with "too old", "too new" etc could be actually put in the exception)